pkg/alerting/init.go

```diff
@@ -6,27 +6,41 @@ import (
 	"fmt"
 	"net/http"
 
+	"github.com/vmihailenco/taskq/v4"
+	"go.uber.org/fx"
+
 	"github.com/uptrace/bunrouter"
+	"github.com/uptrace/opentelemetry-go-extra/otelzap"
 	"github.com/uptrace/uptrace/pkg/bunapp"
+	"github.com/uptrace/uptrace/pkg/bunconf"
 	"github.com/uptrace/uptrace/pkg/httperror"
 	"github.com/uptrace/uptrace/pkg/org"
-	"github.com/vmihailenco/taskq/v4"
 )
 
-func Init(ctx context.Context, app *bunapp.App) {
-	initRouter(ctx, app)
-	initTasks(ctx, app)
-}
-
-func initRouter(ctx context.Context, app *bunapp.App) {
-	middleware := NewMiddleware(app)
-	api := app.InternalAPIV1()
+var Module = fx.Module("alerting",
+	fx.Provide(
+		fx.Private,
+		NewMiddleware,
+		NewAlertHandler,
+		NewMonitorHandler,
+		NewNotifChannelHandler,
+	),
+	fx.Invoke(
+		fx.Annotate(initRouter, fx.ParamTags(`name:"router_internal_apiv1"`)),
+		initTasks,
+	),
+)
 
+func initRouter(
+	api *bunrouter.Group,
+	middleware Middleware,
+	alertHandler *AlertHandler,
+	monitorHandler *MonitorHandler,
+	notifChannelHandler *NotifChannelHandler,
```

Comment:
This probably won't scale well. Maybe move the registration to the corresponding `New...Handler` method? Or create a separate `Register...Handler` function just to register handler routes?